pkg/destroy/aws/aws.go

```diff
@@ -265,6 +265,31 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) ([]string, erro
 	return nil, nil
 }
 
+// findUntaggableResources returns the resources for the cluster that cannot be tagged. Any resource that contains
+// a shared tag will be ignored.
+//
+//	deleted - the resources that have already been deleted. Any resources specified in this set will be ignored.
+func (o *ClusterUninstaller) findUntaggableResources(ctx context.Context, iamClient *iam.IAM, deleted sets.String) (sets.String, error) {
+	resources := sets.NewString()
+	o.Logger.Debug("search for IAM instance profiles")
+	for _, profileType := range []string{"master", "worker", "bootstrap"} {
+		profile := fmt.Sprintf("%s-%s-profile", o.ClusterID, profileType)
+		response, err := iamClient.GetInstanceProfileWithContext(ctx, &iam.GetInstanceProfileInput{InstanceProfileName: &profile})
+		if err != nil {
+			var awsErr awserr.Error
+			if errors.As(err, &awsErr) && awsErr.Code() == iam.ErrCodeNoSuchEntityException {
+				continue
+			}
+			return resources, errors.Wrap(err, "failed to get IAM instance profile")
```

Comment:
Since we are reintroducing this, maybe we can get rid of the deprecated `errors` pkg?
```suggestion
			return resources, fmt.Errorf("failed to get IAM instance profile: %w", err)
```